### ISSUE TITLE
configDebugInfo flag not set right in preConf and confHook

### ISSUE THREAD

**Issue by flip111** [haskell/cabal#6371]
I execute `stack --ghc-options="-g" --no-strip` the result is that my executable has indeed debugging symbols (GDB found them). However in the user hooks `ConfigFlags` data structure  ([link1](https://github.com/haskell/cabal/blob/21762b71a1ac8a331f8ecf05a2e2bd65886d1f45/Cabal/Distribution/Simple/Setup.hs#L277) [link2](https://github.com/haskell/cabal/blob/9dfcece02112766f1e3f3fe7ea2aa79a63b995f0/Cabal/Distribution/Simple/Compiler.hs#L251-L255)) is still set to `NoDebugInfo`. I need to be able to detect if the user (of my library) requested debugging with symbols because then i can send the right flags to cmake to compile the C sources also with debugging symbols.

**Comment by phadej** [comment]
cabal has `cabal build --enable-debug-info` which sets the right flags. `ghc-options` is opaque, and we never try to infer anything from that.

I don't now if `stack` has higher level option to enable debug info, it should have for this to reliably work.

TL;DR it's as wrong as enabling optimization by `--ghc-options=-O`, you'll just confuse every tool.

**Comment by flip111** [comment]
That makes sense. Thanks @phadej
